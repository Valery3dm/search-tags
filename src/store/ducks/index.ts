import { combineReducers } from 'redux';
import { itemReducer } from './itemReducer';

export const rootReducer = combineReducers({
    item: itemReducer
})

export type RootState = ReturnType<typeof rootReducer>
