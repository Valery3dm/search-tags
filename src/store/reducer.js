const initialState = {
  itemsList: [],
  listOfThreeLastItems: [],
  inputItem: ''
};

const FETCH_ITEMS = "FETCH_ITEMS";
const SET_THREE_LAST_ITEMS = "SET_THREE_LAST_ITEMS";
const SET_INPUT_ITEM = "SET_INPUT_ITEM";

const reducer = (state = initialState, action) => {
  switch (action.type) {
    case FETCH_ITEMS:
      return { ...state, itemsList: action.payload };
    case SET_THREE_LAST_ITEMS:
      return { ...state,
         listOfThreeLastItems: [ ...state.listOfThreeLastItems, action.payload]
        };
    case SET_INPUT_ITEM:
      return { ...state, inputItem: action.payload };
    default:
      return state;
  }
};

export default reducer;