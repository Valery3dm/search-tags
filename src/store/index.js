import { applyMiddleware, createStore } from "redux";
import { composeWithDevTools } from "redux-devtools-extension";
import createSagaMiddleware from "redux-saga";

import { duck } from "./ducks/widgets";


const sagaMiddleware = createSagaMiddleware();

const store = createStore(duck.reducers.reducer, composeWithDevTools(applyMiddleware(sagaMiddleware)));

export default store;

sagaMiddleware.run(duck.saga);
