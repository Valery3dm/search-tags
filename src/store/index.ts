import { applyMiddleware, createStore } from 'redux';
import { composeWithDevTools } from 'redux-devtools-extension';
import createSagaMiddleware from 'redux-saga';

import { sagaFetchItemsWatcher } from './saga/sagaFetchItems';
import { itemReducer } from './ducks/itemReducer';

const sagaMiddleware = createSagaMiddleware();

const store = createStore(
  itemReducer,
  composeWithDevTools(applyMiddleware(sagaMiddleware))
);

export default store;

sagaMiddleware.run(sagaFetchItemsWatcher);