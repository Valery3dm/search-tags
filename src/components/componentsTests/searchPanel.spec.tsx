import Enzyme, { mount } from 'enzyme';
import Adapter from '@wojtekmaj/enzyme-adapter-react-17';
import toJson from 'enzyme-to-json';
import { Provider } from 'react-redux';
import store from '../../store';
import { BrowserRouter } from 'react-router-dom';

import { SearchPanel } from '../index';



Enzyme.configure({ adapter: new Adapter() });

describe('Search-Panel component', () => {
    
    it('SearchPanel Snapshot component:', () => {
        const component = mount(
            <Provider store={store}>
                <BrowserRouter>
                    <SearchPanel />
                </BrowserRouter>
            </Provider>
        );
        expect(toJson(component)).toMatchSnapshot();
    });
})
