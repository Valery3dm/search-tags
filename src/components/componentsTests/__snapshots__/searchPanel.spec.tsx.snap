// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Search-Panel component SearchPanel Snapshot component: 1`] = `
<Provider
  store={
    Object {
      "@@observable": [Function],
      "dispatch": [Function],
      "getState": [Function],
      "replaceReducer": [Function],
      "subscribe": [Function],
    }
  }
>
  <BrowserRouter>
    <Router
      history={
        Object {
          "action": "POP",
          "block": [Function],
          "createHref": [Function],
          "go": [Function],
          "goBack": [Function],
          "goForward": [Function],
          "length": 1,
          "listen": [Function],
          "location": Object {
            "hash": "",
            "pathname": "/",
            "search": "",
            "state": undefined,
          },
          "push": [Function],
          "replace": [Function],
        }
      }
    >
      <SearchPanel>
        <Styled(Component)>
          <ForwardRef(InputGroup)
            className="sc-fujyAs kTYqnT"
          >
            <div
              className="sc-fujyAs kTYqnT input-group"
            >
              <FormControl
                aria-describedby="basic-addon2"
                aria-label="Recipient's username"
                onChange={[Function]}
                onKeyPress={[Function]}
                placeholder="type your tag here"
                value=""
              >
                <input
                  aria-describedby="basic-addon2"
                  aria-label="Recipient's username"
                  className="form-control"
                  onChange={[Function]}
                  onKeyPress={[Function]}
                  placeholder="type your tag here"
                  value=""
                />
              </FormControl>
              <InputGroupAppend>
                <div
                  className="input-group-append"
                >
                  <Link
                    to="/"
                  >
                    <LinkAnchor
                      href="/"
                      navigate={[Function]}
                    >
                      <a
                        href="/"
                        onClick={[Function]}
                      >
                        <Button
                          active={false}
                          disabled={false}
                          onClick={[Function]}
                          variant="outline-secondary"
                        >
                          <button
                            className="btn btn-outline-secondary"
                            disabled={false}
                            onClick={[Function]}
                            type="button"
                          >
                            Search
                          </button>
                        </Button>
                      </a>
                    </LinkAnchor>
                  </Link>
                </div>
              </InputGroupAppend>
            </div>
          </ForwardRef(InputGroup)>
        </Styled(Component)>
      </SearchPanel>
    </Router>
  </BrowserRouter>
</Provider>
`;
